opm/output/eclipse/Summary.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_OUTPUT_SUMMARY_HPP
+#define OPM_OUTPUT_SUMMARY_HPP
+
+#include <string>
+#include <vector>
+
+#include <ert/ecl/ecl_sum.h>
+#include <ert/util/ert_unique_ptr.hpp>
+
+#include <opm/output/Wells.hpp>
+
+namespace Opm {
+
+    class EclipseState;
+    class Summary;
+
+namespace out {
+
+class Summary {
+    public:
+        Summary( const EclipseState&, const Opm::Summary& );
```

Comment:
I'm inclined to agree, considering it essentially only creates a configuration *based* on the keywords in the summary section, instead of being a representation of the summary section as the name may indicate.